### ISSUE TITLE
Build and deploy the new site project to dev env.

### ISSUE THREAD

**Issue by dadiorchen** [Greenstand/treetracker-web-map-client#122]
* Because we switched to next.js, the build and deployment would change, previously, we are single page app, and using creat-react-app, so it's about building with CRA and deploy to CDN (static files), now we need build with next.js, and need to deploy to I guess node.js server to handle page rendering. 
* We need to do it in this branch `web-map-site` rather than `master`.
* Please ask in the slack channel if you need special permission.

---
Some hints:
* Please read our [readme](https://github.com/Greenstand/treetracker-web-map-client#treetracker-web) for more information/guide/tutorial.
* Here is [an engineering book](https://greenstand.gitbook.io/engineering/) in Greenstand.
* To know more about our organization, visit our [website](https://greenstand.org).

**Comment by jwgatt** [comment]
Could i be assigned to this?

**Comment by dadiorchen** [comment]
@jwgatt that would be awesome! We still don't have volunteer for us to work on devops, for this job, I suggest you split it into two steps:
1. Just deploy our current code into dev env, please refer to our current workflow for master (this job is for `web-ma-site` branch), https://github.com/Greenstand/treetracker-web-map-client/tree/web-map-site/.github/workflows
2. When @CalDev98 finished the next.js transition #117 , let's consider deploy the next.js server side.

**Comment by jwgatt** [comment]
So should we edit the .yml files in web-ma-site so that they are compatible with the Next.js framework?

**Comment by dadiorchen** [comment]
Do you mean edit the existing action file? I think we should make a new one against the `web-map-site` branch.
As for the next.js, I need your advice on how to deploy it to next.js, @CalDev98 it should be a node.js process running on the server side, right?

**Comment by jwgatt** [comment]
Does this still need to be completed? It seems like the workflow files have been updated.

**Comment by dadiorchen** [comment]
Yes, it is, but I have no idea how to implement it, you know, we are in two branch, master and web-map-site, the former is running on the prod, so the problem is how to maintain the master and deploy the second one to dev env too, also the new branch will use a new way to deploy the app based on next.js, the master is based on react.

**Comment by dadiorchen** [comment]
Any suggestion?

**Comment by jwgatt** [comment]
Seems like workflow files are only picked up from master 
https://stackoverflow.com/questions/63480433/how-to-run-a-github-action-from-a-branch-other-than-master

**Comment by dadiorchen** [comment]
I noticed on the Github, we can chose a branch?
<img width="1440" alt="Screen Shot 2021-10-07 at 11 31 05 AM" src="https://user-images.githubusercontent.com/5744708/136316047-1bffe72b-7ae1-479f-accb-a1dc9203769f.png">

**Comment by dadiorchen** [comment]
And I think it's also acceptable if we add some new workflow into the master branch, but it's for deploying `web-map-site`, right?
